CDCgov/prime-reportstream
# Enhancements to the FHIR Path Tool

This PR makes improvements to the FHIR Path tool.

Test Steps:
1. Check out this branch and build the prime CLI by running `./gradlew fatjar`
2. Run the FHIR path tool by running `./prime fhirpath -i src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir`
3. Type in the following paths in sequence and verify you get to see the contents of the extension
```
resource: 'Bundle.entry.resource.ofType(MessageHeader)'
%resource.source.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/source-software-vendor-org').value
resource = %resource.source.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/source-software-vendor-org').value
%resource
```

## Changes
- Allow to set the %resource by typing ANY of the following:
```
resource : 'Bundle.entry.resource.ofType(MessageHeader)'
resource = 'Bundle.entry.resource.ofType(MessageHeader)'
resource: Bundle.entry.resource.ofType(MessageHeader)
resource= Bundle.entry.resource.ofType(MessageHeader)
```
- Prints out information about non-root resources, but will not walk deeper into a properties to print out other properties (prints out one level of properties)

## Checklist

### Testing

### Process

## Linked Issues
Not part of an issue

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
